app/next-client-app/app/scanreports/[id]/tables/[tableId]/add-concept.tsx

```diff
@@ -0,0 +1,166 @@
+import {
+  getConcept,
+  getConceptFilters,
+  getContentTypeId,
+  addConcept,
+} from "@/api/concepts";
+import { getOmopFields, getOmopTable } from "@/api/omop";
+import { getScanReportConcept, getScanReportTable } from "@/api/scanreports";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { m_allowed_tables } from "@/constants/concepts";
+import { ApiError } from "@/lib/api/error";
+import { mapConceptToOmopField, saveMappingRules } from "@/lib/concept-utils";
+import { Form, Formik } from "formik";
+import { toast } from "sonner";
+import { objToQuery } from "@/lib/client-utils";
+
+interface AddConceptProps {
+  id: number;
+  tableId: string;
+}
+
+export default function AddConcept({ id, tableId }: AddConceptProps) {
+  const typeNameQuery = objToQuery({
+    type_name: "scanreportfield",
+  });
+  const objectQuery = objToQuery({
+    object_id: id,
+  });
+
+  const handleError = (error: any, message: string) => {
+    const errorObj = JSON.parse((error as ApiError).message);
+    toast.error(`${message} Error: ${errorObj.detail}`);
+    console.error(error);
+  };
+
+  const handleSubmit = async (conceptCode: number) => {
+    try {
+      const table = await getScanReportTable(tableId);
+      const concept = await getConcept(conceptCode);
+      const domain = concept?.domain_id.toLocaleLowerCase() ?? "";
+      const fields = await getOmopFields();
+      const contentType = await getContentTypeId(typeNameQuery);
+      // set the error message depending on which value is missing
+      if (!table.person_id || !table.date_event) {
```

Comment:
As mentioned, I think we should move to getting the API to do this validation for us, and remove it all from Next.

What I'm keen to hear - is that given your closer work on it and better understanding than me, is this plausible to do, and if it would simplify the code, to what level?